src/DotVVM.Framework.Tests.Common/ResourceManagement/ResourceRepositoryTests.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using DotVVM.Framework.Configuration;
+using DotVVM.Framework.ResourceManagement;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace DotVVM.Framework.Tests.Common.ResourceManagement
+{
+    [TestClass]
+    public class ResourceRepositoryTests
+    {
+        [TestMethod]
+        public void ResourceRepository_CyclicDependency_Throws()
```

Comment:
IMHO this unit test doesn't work as you intended.
If it throws or not the unit test will succeed.
If you except to throw exception you should use something like this: https://docs.microsoft.com/cs-cz/dotnet/api/microsoft.visualstudio.testtools.unittesting.assert.throwsexception?view=mstest-net-1.2.0

Reply:
Good point. Fixed. Thank you. :)